TulparAudio/internal/source/SourceCollection.cpp

```diff
@@ -321,7 +321,16 @@ audio::Buffer SourceCollection::GetSourceActiveBuffer(SourceHandle source) const
         case audio::Source::Type::Streaming:
         {
             std::vector<audio::Buffer> queue = GetSourceQueuedBuffers(source);
-            buffer = queue[GetSourceQueueIndex(source)];
+            uint32_t const index = GetSourceQueueIndex(source);
+
+            if (index < queue.size())
+            {
+                buffer = queue[index];
+            }
+            else
+            {
+                buffer = queue.at(0);
```

Comment:
Is this queue guaranteed to be at least of size 1 or otherwise should user code handle a possible exception?

Reply:
Ultimately type of current source is provided by OpenAL. To avoid relying on some internal mechanisms I would rather check for `queue.empty()` prior to requesting the index.